pandas/core/indexes/datetimelike.py

```diff
@@ -214,7 +248,8 @@ def _summary(self, name=None) -> str:
     def _can_partial_date_slice(self, reso: Resolution) -> bool:
         # e.g. test_getitem_setitem_periodindex
         # History of conversation GH#3452, GH#3931, GH#2369, GH#14826
-        return reso > self._resolution_obj
+        # error: Unsupported operand types for < ("str" and "Resolution")
```

Comment:
it thinks that reso is a str?

Reply:
must have copied the wrong error message. The "correct" error is

> error: Unsupported left operand type for > ("Resolution")  [operator] 
> note: Right operand is of type "Optional[Resolution]"

Reply:
I added a breakpoint, both of them are Resolution objects